pkg/proxy/winkernel/proxier.go

```diff
@@ -1131,7 +1138,7 @@ func (proxier *Proxier) syncProxyRules() {
 
 		hnsLoadBalancer, err := hns.getLoadBalancer(
 			hnsEndpoints,
-			loadBalancerFlags{isDSR: proxier.isDSR, isIPv6: proxier.isIPv6Mode, sessionAffinity: sessionAffinityClientIP},
+			loadBalancerFlags{isDSR: proxier.isDSR || svcInfo.dsr, isIPv6: proxier.isIPv6Mode, sessionAffinity: sessionAffinityClientIP},
```

Comment:
I believe that we should not be modifying this loadbalancer as externalTrafficPolicy: Local only affects the nodeport loadbalancer. Same for below.